src/network/grpc/client.rs

```diff
@@ -16,10 +16,12 @@ where
     B: NetworkBlockConfig,
 {
     info!("connecting to subscription peer {}", peer.connection);
+    info!("address: {}", peer.address());
     let peer = grpc_peer::TcpPeer::new(*peer.address());
     let addr = peer.addr().clone();
+    let origin: http::Uri = format!("http://{}", addr).parse().unwrap();
```

Comment:
Better do it with `uri::Builder`, we have all the parts, no need to format and parse.